security/pkg/nodeagent/plugin/providers/google/stsclient/stsclient.go

```diff
@@ -106,6 +106,7 @@ func (p *SecureTokenServiceExchanger) requestWithRetry(reqBytes []byte) ([]byte,
 			break
 		}
 		monitoring.NumOutgoingRetries.With(monitoring.RequestType.Value(monitoring.TokenExchange)).Increment()
+		stsClientLog.Errorf("token exchange request failed: status code %v", resp.StatusCode)
```

Comment:
can we make it conditional on if stsClientLog.DebugEnabled? it will be confusing to get 2 logs for one event IMO. We do this in other places

Reply:
Revised accordingly.